compiler/src/dmd/backend/ptrntab.d

```diff
@@ -5833,9 +5833,9 @@ extern (C++) OP *asm_op_lookup(const(char)* s)
 @trusted
 private int binary(const(char)* p, const OP[] table)
 {
-    uint low = 0;
+    int low = 0;
     char cp = *p;
-    uint high = cast(uint)(table.length) - 1;
+    int high = cast(int)(table.length) - 1;
```

Comment:
At line 5848 (github is annoying and I can't comment directly there) `high = mid - 1` . When mid is 0, BOOM!

Reply:
```suggestion
    uint high = cast(uint)table.length;
```
Wouldn't this work?

Reply:
No, the problem appears when you are searching for an element that does not exist

Reply:
That should be `high = mid;` instead of `high = mid - 1;`.